src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -617,6 +633,230 @@ BoundExpression convertArrayElement(BoundExpression element, TypeSymbol elementT
+                return BindToNaturalType(element, diagnostics, reportNoTargetType: false);
+            }
+
+            BoundExpression convertDictionaryElement(
+                BoundCollectionLiteralDictionaryElement dictionaryElement,
+                BoundObjectOrCollectionValuePlaceholder receiver,
+                PropertySymbol indexer,
+                NamedTypeSymbol targetType,
+                BindingDiagnosticBag diagnostics)
+            {
+                var typeArgs = targetType.TypeArgumentsWithAnnotationsNoUseSiteDiagnostics;
+                Debug.Assert(typeArgs.Length == 2);
+
+                var key = convertExpression(dictionaryElement.Key, typeArgs[0].Type, diagnostics);
+                var value = convertExpression(dictionaryElement.Value, typeArgs[1].Type, diagnostics);
+
+                return dictionaryElement.Update(
+                    key,
+                    value,
+                    setValue(dictionaryElement.Syntax, receiver, indexer, key, value));
+            }
+
+            BoundExpression convertSpreadElement(
+                SyntaxNode syntax,
+                BoundCollectionLiteralSpreadElement element,
+                BoundObjectOrCollectionValuePlaceholder implicitReceiver,
+                PropertySymbol indexer,
+                NamedTypeSymbol targetType,
+                BindingDiagnosticBag diagnostics)
+            {
+                var typeArgs = targetType.TypeArgumentsWithAnnotationsNoUseSiteDiagnostics;
+                Debug.Assert(typeArgs.Length == 2);
+
+                var enumeratorInfo = element.EnumeratorInfoOpt;
+                if (enumeratorInfo is null)
+                {
+                    // PROTOTYPE: What is the case? Have we reported an error?
+                    return element;
+                }
+
+                Debug.Assert(enumeratorInfo.ElementType is { }); // ElementType is set always, even for IEnumerable.
+
+                if (!enumeratorInfo.ElementType.OriginalDefinition.Equals(Compilation.GetWellKnownType(WellKnownType.System_Collections_Generic_KeyValuePair_KV), TypeCompareKind.AllIgnoreOptions))
+                {
+                    diagnostics.Add(ErrorCode.ERR_CollectionLiteralDictionarySpreadUnexpectedType, syntax);
+                    return BadExpression(syntax);
+                }
+
+                var addElementPlaceholder = new BoundValuePlaceholder(syntax, enumeratorInfo.ElementType);
+
+                var key = convertExpression(
+                    getPropertyValue(syntax, addElementPlaceholder, WellKnownMember.System_Collections_Generic_KeyValuePair_KV__get_Key, diagnostics),
+                    typeArgs[0].Type,
+                    diagnostics);
+                var value = convertExpression(
+                    getPropertyValue(syntax, addElementPlaceholder, WellKnownMember.System_Collections_Generic_KeyValuePair_KV__get_Value, diagnostics),
+                    typeArgs[1].Type,
+                    diagnostics);
```

Comment:
just realize dthe spec doesn't cover this case.  It says the translation is: 

![image](https://github.com/dotnet/roslyn/assets/4564579/8cf7fa96-2702-4727-8264-64438373fc94)

Where it needs to call out the KVP understanding in the translation.